packages/node/src/indexer/unfinalizedBlocks.service.spec.ts

```diff
@@ -0,0 +1,194 @@
+    expect(
+      (unfinalizedBlocksService as any).unfinalizedBlocks[100],
+    ).toBeDefined();
+    expect(
+      (unfinalizedBlocksService as any).unfinalizedBlocks[80],
+    ).toBeUndefined();
+    expect(
+      (unfinalizedBlocksService as any).unfinalizedBlocks[50],
+    ).toBeUndefined();
+    unfinalizedBlocksService.removeFinalized(150);
+    expect(
+      (unfinalizedBlocksService as any).unfinalizedBlocks[100],
+    ).toBeUndefined();
+    expect(
+      (unfinalizedBlocksService as any).unfinalizedBlocks[120],
+    ).toBeUndefined();
+
+    // unfinalizedBlocksService.getClosestRecord()
+  });
+
+  it('can validate best block if finalized block can be found in its record', async () => {
+    unfinalizedBlocksService.storeUnfinalizedBlock(50, '0x50abcd');
+    unfinalizedBlocksService.storeUnfinalizedBlock(80, '0x80abcd');
+    unfinalizedBlocksService.storeUnfinalizedBlock(100, '0x100abc');
+    unfinalizedBlocksService.storeUnfinalizedBlock(120, '0x120abc');
+    unfinalizedBlocksService.registerFinalizedBlock({
+      block: { header: { number: { toNumber: () => 100 } } },
+      hash: { toHex: () => '0x100abc' },
+    } as unknown as SignedBlock);
+    await expect(
+      unfinalizedBlocksService.validateUnfinalizedBlocks(),
+    ).resolves.toBeTruthy();
+    unfinalizedBlocksService.registerFinalizedBlock({
+      block: { header: { number: { toNumber: () => 120 } } },
+      hash: { toHex: () => '0x120xyz' },
+    } as unknown as SignedBlock);
+    await expect(
+      unfinalizedBlocksService.validateUnfinalizedBlocks(),
+    ).resolves.toBeFalsy();
+    // unfinalizedBlocksService.getClosestRecord()
+  });
+
+  it('can validate best block if finalized block can NOT be found in its record', async () => {
+    unfinalizedBlocksService.storeUnfinalizedBlock(50, '0x50abcd');
+    unfinalizedBlocksService.storeUnfinalizedBlock(80, '0x80abcd');
+    unfinalizedBlocksService.storeUnfinalizedBlock(100, '0x100abc');
+    unfinalizedBlocksService.storeUnfinalizedBlock(120, '0x123456');
+    unfinalizedBlocksService.registerFinalizedBlock({
+      block: { header: { number: { toNumber: () => 150 } } },
+      hash: { toHex: () => '0x100abc' },
+    } as unknown as SignedBlock);
+    await expect(
+      unfinalizedBlocksService.validateUnfinalizedBlocks(),
+    ).resolves.toBeTruthy();
+  });
+
+  it('invalid best block will be removed', async () => {
+    unfinalizedBlocksService.storeUnfinalizedBlock(100, '0x100abc');
+    unfinalizedBlocksService.storeUnfinalizedBlock(120, '0x120abc');
+    unfinalizedBlocksService.registerFinalizedBlock({
+      block: { header: { number: { toNumber: () => 150 } } },
+      hash: { toHex: () => '0x100abc' },
+    } as unknown as SignedBlock);
+    // blockHash mocked value is `0x123456`
+    await expect(
+      unfinalizedBlocksService.validateUnfinalizedBlocks(),
+    ).resolves.toBeFalsy();
+    expect(unfinalizedBlocksService.unfinalizedBlock(120)).toBeUndefined();
```

Comment:
Which function removed block 120? I don't think it is `registerFinalizedBlock` or `validateUnfinalizedBlocks`